packages/angular_devkit/core/src/json/schema/registry_spec.ts

```diff
@@ -251,4 +251,56 @@ describe('CoreSchemaRegistry', () => {
       )
       .subscribe(done, done.fail);
   });
+
+  it('supports smart defaults', done => {
+    const registry = new CoreSchemaRegistry();
+    const data: any = {
+      arr: [{}],
+    };
+
+    registry.addSmartDefaultProvider('test', () => true);
+    registry.addSmartDefaultProvider('test2', (x) => x['blue']);
```

Comment:
Can you add a test for the absolute pointer? I think it's supported from the code, but not tested. 

Relative pointers are not supported, right?

Reply:
Discussed offline.